src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -293,5 +344,73 @@ std::string ObjectCacheStore::get_cache_file_path(std::string cache_file_name,
   return m_cache_root_dir + cache_file_dir + cache_file_name;
 }
 
+void ObjectCacheStore::handle_throttle_ready(int ret, void* ctx, uint64_t flag) {
+  m_io_throttled.store(false);
+}
+
+bool ObjectCacheStore::take_token_from_throttle(uint64_t object_size,
+                                                uint64_t object_num) {
+  bool got_token = true;
+  
+  if (m_io_throttled.load() == true) {
+    return true;
+  }
+
+  if (got_token && (m_qos_enabled_flag & ROC_QOS_IOPS_THROTTLE)) {
+    got_token = 
+      m_throttles[ROC_QOS_IOPS_THROTTLE]->get<ObjectCacheStore, void,
+       &ObjectCacheStore::handle_throttle_ready>(object_size, this, nullptr, 0);
+  } 
+
+  if (got_token && (m_qos_enabled_flag & ROC_QOS_BPS_THROTTLE)) {
+    got_token =
```

Comment:
What if both throttles are enabled and you are blocked by the first one? It seems like this throttle won't track BPS.